translatable_proc/src/macro_generation/translation.rs

```diff
@@ -0,0 +1,111 @@
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{ToTokens, quote};
+use thiserror::Error;
+use translatable_shared::handle_macro_result;
+use translatable_shared::macros::collections::map_to_tokens;
+use translatable_shared::misc::language::Language;
+
+use crate::data::translations::load_translations;
+use crate::macro_input::input_type::InputType;
+use crate::macro_input::translation::TranslationMacroArgs;
+
+#[derive(Error, Debug)]
+enum MacroCompileError {
+    #[error("The path '{0}' could not be found")]
+    PathNotFound(String),
+
+    #[error("The language '{0:?}' ('{0:#}') is not available for the path '{1}'")]
+    LanguageNotAvailable(Language, String),
+}
+
+pub fn translation_macro(input: TranslationMacroArgs) -> TokenStream2 {
+    let translations = handle_macro_result!(load_translations());
+
+    let template_replacements =
+        input.replacements().iter().map(|(key, value)| quote! { #key = #value });
+
+    if let InputType::Static(language) = input.language() {
+        if let InputType::Static(path) = input.path() {
+            let static_path_display = path.join("::");
+
+            let translation_object = translations
+                .find_path(&path)
```

Comment:
Since `path` is already a reference, there is no need to make another reference.

```suggestion
                .find_path(path)
```